packages/server/lib/controllers/v1/getProvidersJSON.ts

```diff
@@ -0,0 +1,13 @@
+import { getProviders } from '@nangohq/shared';
+import { asyncWrapper } from '../../utils/asyncWrapper.js';
+
+let providersJSON: string | undefined = undefined;
+
+export const getProvidersJSON = asyncWrapper<any, any>((_, res) => {
+    if (!providersJSON) {
+        providersJSON = JSON.stringify(getProviders());
+    }
```

Comment:
the providers content is already cached in memory. Is it worth caching the json as well? How slow/fast is the stringifying?

Reply:
I think you mean caching in `shared` like we do with the providers data itself. This is caching the data in memory in the controller here. 

The reason I put this here is that I wasn't sure if we wanted to have that extra chunk of memory always in use by anything that uses providers from shared. It's currently about 190kb, so it's not like it's too big of a deal, but I figured we didn't want side effects that like that consistently leaking out of `shared`.

It takes around 500-700 microseconds to stringify the JSON right now and only has to happen the first time this controller is called, so I don't think it's too big of a deal.

Reply:
up to you. It doesn't seem to be a big deal either way. 🚢

Reply:
All for caching, but if you do it, you will need to deal with cache invalidation in the next PR 
Here you have no way to clear the cache easily

Reply:
server restart/deployment would "invalidate" the cache.

Reply:
ah yes true we rely on the deploy, then fine 👌🏻